File: utils/VectorEnginePlugin.ref/pares/funciones/primFillRGBA.c
/* VectorEnginePlugin>>#fillR:g:b:a: */
EXPORT(sqInt)
primFillRGBA(void)
{
	double a;
	double b;
	double g;
	double r;

	if (!((isFloatObject(stackValue(3)))
		 && ((isFloatObject(stackValue(2)))
		 && ((isFloatObject(stackValue(1)))
		 && (isFloatObject(stackValue(0))))))) {
		return primitiveFailFor(PrimErrBadArgument);
	}
	r = stackFloatValue(3);
	g = stackFloatValue(2);
	b = stackFloatValue(1);
	a = stackFloatValue(0);
	fillR = r * 255.0;
	fillG = g * 255.0;
	fillB = b * 255.0;
	fillA = a;
	if (!(failed())) {
		pop(4);
	}
	return null;
}

File: utils/VectorEnginePlugin.ref/pares/funciones/primFillRGBA.js
function primFillRGBA(argCount) {
    if (!(isFloatObject(stackValue(3)) && isFloatObject(stackValue(2))
        && isFloatObject(stackValue(1)) && isFloatObject(stackValue(0)))) {
        primitiveFailFor(PrimErrBadArgument); return false;
    }
    fillR = stackFloatValue(3) * 255.0;
    fillG = stackFloatValue(2) * 255.0;
    fillB = stackFloatValue(1) * 255.0;
    fillA = stackFloatValue(0);
    if (!failed()) pop(4);
    return !failed();
}
